zerver/lib/push_notifications.py

```diff
@@ -735,7 +748,28 @@ def push_notifications_configured() -> bool:
 
 
 def initialize_push_notifications() -> None:
+    """Called during startup of the push notifications worker to check
+    whether we expect mobile push notifications to work on this server
+    and update state accordingly.
+    """
+
+    # TODO: Think about do_create_realm code path. Might be OK because
+    # one needs to register the realm.
+    # TODO: Handle initializing these values when importing realms.
```

Comment:
The answer for both of these is that https://github.com/zulip/zulip/pull/28008 should be extended to process the `realms` response, I think.

Reply:
So #28008 was merged. I think basically we need to extend a67dd6dc1fb4b9946ff50fb57e08eaf99b3cf6cf to look at the 

A plausible theory is that we should just have `send_realms_only_to_push_bouncer` be responsible for doing this update code. @mateuszmandera thoughts? 
```
        for realm_uuid, data in realms.items():                                                                         
            realm = Realm.objects.get(uuid=realm_uuid)                                                                  
            do_set_realm_property(                                                                                      
                realm, "push_notifications_enabled", data["can_push"], acting_user=None                                 
            )                                                                                                           
            do_set_push_notifications_enabled_end_timestamp(                                                            
                realm, data["expected_end_timestamp"], acting_user=None                                                 
            )                                                                                                           
```

I guess actually we probably want to process the response regardless of whether we sent only realms or sent realms plus other data; I think the short story is we should turn this block of code into a function that lives in `zerver/lib/remote_server.py` so it's easy to call from all the places we need it.

I think it's only possibly an improvement to do this update every single time, rather than just when starting up the push notifications process. (And in fact, it'll be essential for correctness when the values change that we do this in other code paths).

Reply:
Let's track this in https://github.com/zulip/zulip/issues/28031.